### DISCUSSION TITLE
What does this grammar mean

### DISCUSSION THREAD

**Discussion by myNameIsDu** [babel/babel#12733] [Q&A] [answered]
I found this grammar when I read the source code of babel-parser:

![image](https://user-images.githubusercontent.com/53305259/106500094-5ffec380-64fc-11eb-8f54-ba03b488be07.png)

In the switch statement:

![image](https://user-images.githubusercontent.com/53305259/106500275-94727f80-64fc-11eb-9ff1-c0211ac8de8e.png)

I found them meaningless and ignored, But why write it like this, Will the prompt terminate the while statement?

**Comment by nicolo-ribaudo** [comment, marked as answer]
Yes, without that label `break` would terminate the `switch` statement instead of the loop.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/label
